File: lib/RubyGL/Native/src/Input.c
#include "Input.h"

int initInput()
{
	return SDL_InitSubSystem(SDL_INIT_EVENTS);
}

void quitInput()
{
	SDL_QuitSubSystem(SDL_INIT_EVENTS);
}

void addEventWatch(SDL_EventFilter filter, void* userdata)
{
	SDL_AddEventWatch(filter, userdata);
}

void pumpEvents(void)
{
	SDL_PumpEvents();
}

File: lib/RubyGL/Native/input.rb
require 'ffi'

module RubyGL::Native
    class EventType < FFI::Union
        
    end

    attach_function :initInput, [], :int
    attach_function :quitInput, [], :void
    attach_function :addEventWatch, [:pointer, :pointer], :void
    attach_function :deleteEventWatch, [:pointer, :pointer], :void
    attach_function :pumpEvents, [], :void
end